collect_app/src/test/java/org/odk/collect/android/formentry/audit/FormSaveViewModelTest.java

```diff
@@ -325,6 +338,39 @@ public void whenFormSaverFinishes_isSaving_returnsFalse() {
         assertThat(viewModel.isSaving(), equalTo(false));
     }
 
+    @Test
+    public void whenFormSaverFinishes_mediaManagerIsCleared() {
+        viewModel.markOriginalFileOrDelete("index", "blah");
+        viewModel.replaceRecentFileForQuestion("index", "blah");
+
+        viewModel.saveForm(Uri.parse("file://form"), true, "", true);
+        whenFormSaverFinishes(SaveFormToDisk.SAVED);
+
+        Map<String, String> originalFiles = savedStateHandle.get(FormSaveViewModel.ORIGINAL_FILES);
+        Map<String, String> recentFiles = savedStateHandle.get(FormSaveViewModel.RECENT_FILES);
+
+        assertThat(originalFiles.isEmpty(), equalTo(true));
```

Comment:
Instead of verifying the files here I think this test should run `saveForm` again and check that `deleteImageFileFromMediaProvider` is only called the first time. Does that make sense? Again, this is to test the intended behaviour rather than an implementation detail of that behaviour.

Reply:
To verify calls to `deleteImageFileFromMediaProvider()`, I should use actual `new DisFormSaver()` instead of `FormSaver`, right?

Reply:
Oh I just meant like you did in the other test and call `saveForm` on the view model. Does that make sense. The test is to verify that if we call save once and then call save again then we don't try and delete the files from the first save. That's my understanding anyway. Does that make sense?

Reply:
So pseudo code would be:

* set up files to be deleted/saved
* call save on view model
* check files are saved/deleted correctly (using mock of `MediaUtils`)
* call save again
* check no files are saved/deleted (you can pass `once()` to `verify` to check it method has only been called once)

Reply:
Hi! the unit test `saveForm_runsWith_correctFiles()` is failing. Sorry I meant to name it `saveForm_savesCorrectFiles()`. I confirmed the failure is bacause of calling `clearMediaFiles()` in OnComplete, but that should not effect the FakeFormSaver, right?